tests/trainer/test_trainer.py

```diff
@@ -1179,3 +1180,22 @@ def test_row_log_interval(log_metrics_mock, tmpdir, train_batches, max_steps, lo
     trainer.fit(model)
     expected_calls = [call(metrics=ANY, step=s) for s in range(log_interval - 1, max_steps, log_interval)]
     log_metrics_mock.assert_has_calls(expected_calls)
+
+
+@pytest.mark.parametrize(['profiler', 'expected'], [
+    (None, PassThroughProfiler),
+    (SimpleProfiler(), SimpleProfiler),
+    (AdvancedProfiler(), AdvancedProfiler),
+    ('simple', SimpleProfiler),
+    ('advanced', AdvancedProfiler),
+])
+def test_trainer_profiler_correct_args(profiler, expected):
+    kwargs = {'profiler': profiler} if profiler is not None else {}
```

Comment:
Thanks for the suggestions @rohitgr7. I kept the `kwargs` here since that also checks that the `profiler` argument is correctly defaulted to `None` in case nothing is passed. Also, your suggestion for this is no longer here I think it's because I force pushed an ammend (sorry :)).